### DISCUSSION TITLE
Toggle show/hidden for left click on tray icon

### DISCUSSION THREAD

**Discussion by cassepipe** [transmission/transmission#5316] [Ideas]
On Cinnamon, left click and right click both show the tray icon context menu.

It'd be nice if instead a left click on the tray icon would show/hide Transmission.

The current way now is a bit inefficient for such a basic action : two clicks, one on the icon the other on the "Show Tranmission"

Furthermore, when you minimize the window  from the window decoration, the 
Show Transmission" checkbox remains checked even though the window is hidden so that half of the time it is out of sync with the current state of the window.

I don't know what it the situation on other DEs but I suspect it is the same.

What do you think ?

**Comment by ckerr** [comment]
I'm not really sure what to do about any tray icon requests since it seems that tray icons aren't actually supported in GTK4. See [this page](https://discourse.gnome.org/t/what-to-use-instead-of-statusicon-in-gtk4-to-display-the-icon-in-the-system-tray/7175) for some background discussion from GNOME.

We probably ought to just leave it alone until / unless upstream provides a clear path forward.

**Reply by cassepipe** [reply]
I wouldn't expect the people over there to provide any clear path forward since apparently from what I read they don't like tray icons and think it's a mess.

I wonder though, how are you supposed to know which apps are running in the background in GNOME ? I don't know about you, but I care about what apps I ran are still running.

I am using Cinnamon on Fedora and surely *something* has been implemented since I do have a tray icon and its menu. Is that something bolted on by the Fedora packager(s) ?

**Reply by cassepipe** [reply]
By the way, one of them cites Cinnamon as being a wayland desktop which I am pretty sure it is not

**Reply by ckerr** [reply]
there is upstream work being done on background apps @ https://gitlab.gnome.org/Teams/Design/os-mockups/-/issues/191 so I don't think all hope is lost :smile_cat:
